include/mppi_controller/motion_models.hpp

```diff
@@ -46,22 +47,60 @@ class MotionModel {
    */
   virtual ~MotionModel() = default;
 
+  /**
+    * @brief Initialize motion model on bringup and set required variables
+    * @param control_constraints Constraints on control
+    * @param model_dt duration of a time step
+    */
+  void initialize(const models::ControlConstraints & control_constraints, float model_dt)
+  {
+    control_constraints_ = control_constraints;
+    model_dt_ = model_dt;
+  }
+
   /**
    * @brief With input velocities, find the vehicle's output velocities
    * @param state Contains control velocities to use to populate vehicle
    * velocities
    */
   virtual void predict(models::State& state) {
-    using namespace xt::placeholders;  // NOLINT
-    xt::noalias(xt::view(state.vx, xt::all(), xt::range(1, _))) =
-        xt::view(state.cvx, xt::all(), xt::range(0, -1));
-
-    xt::noalias(xt::view(state.wz, xt::all(), xt::range(1, _))) =
-        xt::view(state.cwz, xt::all(), xt::range(0, -1));
-
-    if (isHolonomic()) {
-      xt::noalias(xt::view(state.vy, xt::all(), xt::range(1, _))) =
-          xt::view(state.cvy, xt::all(), xt::range(0, -1));
+    const bool is_holo = isHolonomic();
+    const double max_vel_trans = control_constraints_.max_vel_trans;
+    for (unsigned int i = 0; i != state.vx.shape(0); i++) {
```

Comment:
Q: can't we do colum-major access as they do with eigen:
https://github.com/ros-navigation/navigation2/blob/main/nav2_mppi_controller/include/nav2_mppi_controller/motion_models.hpp#L83
?
(invert i with j)